### ISSUE TITLE
SASL cannot be used if nick is already in use or having services lock

### ISSUE THREAD

**Issue by Mikaela** [prawnsalad/KiwiIRC#674]
KiwiIRC uses the nickname as SASL username which requires the nick to be not in use and not having services lock (because of someone else failing to identify in enforcetime). This means that I am unable to identify using SASL so I must not use some other authentication mechanism and change my nick.

This other authentication method with average user is `/msg nickserv identify username password` which is also problematic;
* This happens a lot later than SASL
* I am affected by potential enforce/enforcetime
* There can be server misconfiguration allowing malicious user to `nick NickServ` and receiving my password. While this sounds like something that shouldn't happen, this has actually happened multiple times even on freenode.

**Comment by Mikaela** [comment]
Ways to fix that I see:

* have different box for username
* automatically attempt nicks like nick_, nick__, nick2, nick3 etc. until some nick is usable and I can connect while being identified using SASL.

**Comment by MrBenC** [comment]
I agree with the suggestion of a different username box.  If the username box is empty to try to use the desired nick as the SASL username as a fallback.

Ben

**Comment by Mikaela** [comment]
From what I understood adding separate box is impossible while trying to keep it simple so the solution must be:

> automatically attempt nicks like nick_, nick__, nick2, nick3 etc. until some nick is usable and I can connect while being identified using SASL.

If I have understood correctly, you first identify with SASL and then send NICK and server replies that the NICK is already in use and lets you send more NICKs which hopefully aren't in use.
